### PR TITLE
Chat - File does not appear with strikethrough style when uploaded offline and deleted

### PR DESCRIPTION
**Pull request by Krishna2323**
### Fixed Issues
$ https://github.com/Expensify/App/issues/46616
PROPOSAL: https://github.com/Expensify/App/issues/46616#issuecomment-2263047602

### Tests

1. Go to any chat
2. Upload a file (format: .txt/.html/.zip/)
3. Upload image & video
4. Go offline
5. Right click on the file
6. Click Delete comment
7. Verify an overlay with trash icon is shown for deleted video & image and the filename is crossed for the other file. (format: .txt/.html/.zip/)
8. Stay offline and repeat step 2 to 7

- [x] Verify that no errors appear in the JS console

### Offline tests
1. Go to any chat
2. Upload a file (format: .txt/.html/.zip/)
3. Upload image & video
4. Go offline
5. Right click on the file
6. Click Delete comment
7. Verify an overlay with trash icon is shown for deleted video & image and the filename is crossed for the other file. (format: .txt/.html/.zip/)
8. Stay offline and repeat step 2 to 7

### QA Steps
1. Go to any chat
2. Upload a file (format: .txt/.html/.zip/)
3. Upload image & video
4. Go offline
5. Right click on the file
6. Click Delete comment
7. Verify an overlay with trash icon is shown for deleted video & image and the filename is crossed for the other file. (format: .txt/.html/.zip/)
8. Stay offline and repeat step 2 to 7

- [x] Verify that no errors appear in the JS console

### PR Author Checklist

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android: Native
    - [x] Android: mWeb Chrome
    - [x] iOS: Native
    - [x] iOS: mWeb Safari
    - [x] MacOS: Chrome / Safari
    - [x] MacOS: Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>

https://github.com/user-attachments/assets/fa5edcd0-f80a-4b53-b64f-a9aa89c34268

</details>

<details>
<summary>Android: mWeb Chrome</summary>

https://github.com/user-attachments/assets/0411ffee-3a8a-4264-a274-ea7dda3d2db5

</details>

<details>
<summary>iOS: Native</summary>

https://github.com/user-attachments/assets/3bebe285-1a6f-46f9-a4ff-916a39b651b3

</details>

<details>
<summary>iOS: mWeb Safari</summary>

https://github.com/user-attachments/assets/ef7e805c-6b6f-41aa-ab34-502d58c765e3

</details>

<details>
<summary>MacOS: Chrome / Safari</summary>

https://github.com/user-attachments/assets/ed2e450a-920d-41bd-a8d1-c3cf82cb831f

</details>

<details>
<summary>MacOS: Desktop</summary>

https://github.com/user-attachments/assets/6efb8d68-d709-429e-8a35-aac00d041514

</details>

### CODE REVIEW

**Comment by dubielzyk-expensify** [comment]
There seems to be something weird happening when you go into offline mode and it immediately showing this before you've even deleted anything:
CleanShot 2024-08-28 at 12 16 08@2x

The trash icon should only appear when something is deleted in offline mode

**Comment by shawnborton** [comment]
Yup, I'm also curious about what Jon is mentioning.

**Comment by dannymcclain** [comment]
Same. Other than that weirdness Jon pointed out I think it's looking good.

**Comment by dubielzyk-expensify** [comment]
Thanks for providing video for light mode 😄 

Is it using the icon color variable? Cause it looks to be the same icon color as in dark mode which is a darker than we want. In light mode it should appear a bit lighter. But it could be the recording screwing up with me here.

**Comment by shawnborton** [comment]
I _think_ the video might be correct? I think the color for icons is darker in light mode, and lighter in dark mode.

**Comment by dannymcclain** [comment]
Haha, I'm with both of you—the icon color in light mode doesn't quite look right to me 🤨

Here's a Figma screenshot. The icon is just using our standard `icons` color variable.

CleanShot 2024-08-29 at 09 01 32@2x

**Comment by dubielzyk-expensify** [comment]
Hmm. I would've assumed we applied the opacity only on the image and not the whole container including the icon. 

Keen to hear if @Expensify/design is okay with this or if we should figure out a better way. I definitely wouldn't wanna brighten the colors as we wanna be consistently using our color variables here

**Comment by dannymcclain** [comment]
Yeah I definitely want to just use the variables as they are.

>I would've assumed we applied the opacity only on the image and not the whole container including the icon.

Yeah I agree, though I think the _whole entire post_ gets reduced opacity (even the name/avatar/timestamp) so I think if we're using the right variables I'm fine with how this is looking.

**Comment by dubielzyk-expensify** [comment]
> so I think if we're using the right variables I'm fine with how this is looking.

SGTM. Let's push ahead unless @shawnborton thinks otherwise

**Comment by shawnborton** [comment]
Yup same, all good with me!

**Comment by mananjadhav** [comment]
I couldn't find the solution for the Android bug. I am still working on this.

**Comment by mananjadhav** [comment]
yeah let's do that. @neil-marcellini does that work? @Krishna2323 can you please merge conflicts?

**Comment by mananjadhav** [comment]
@Krishna2323 I think we'll need to fix the eslint issues even if they're unrelated to the PR.  It is because we're touching those files and the new lint rules are failing.

@neil-marcellini can you please confirm?

File: src/components/HTMLEngineProvider/HTMLRenderers/VideoRenderer.tsx [hunk 1]
```diff
@@ -25,6 +25,8 @@ function VideoRenderer({tnode, key}: VideoRendererProps) {
     const height = Number(htmlAttribs[CONST.ATTACHMENT_THUMBNAIL_HEIGHT_ATTRIBUTE]);
     const duration = Number(htmlAttribs[CONST.ATTACHMENT_DURATION_ATTRIBUTE]);
     const currentReportIDValue = useCurrentReportID();
+    const parentStyle = tnode.parent?.styles?.nativeTextRet ?? {};
```
Comment (mananjadhav) [inline src/components/HTMLEngineProvider/HTMLRenderers/VideoRenderer.tsx:28]: I can see this code being repeated quite a few times, can we create a util for this?

File: src/components/ThumbnailImage.tsx [hunk 2]
```diff
@@ -55,6 +56,9 @@ type ThumbnailImageProps = {
 
     /** The object position of image */
     objectPosition?: ImageObjectPosition;
+
+    /**  Whether the image is deleted */
```
Comment (mananjadhav) [inline src/components/ThumbnailImage.tsx:60]:
```suggestion
    /** Whether the image is deleted */
```

File: src/components/Attachments/AttachmentView/index.tsx [hunk 3]
```diff
@@ -77,6 +77,9 @@ type AttachmentViewProps = AttachmentViewOnyxProps &
 
         /* Flag indicating whether the attachment has been uploaded. */
         isUploaded?: boolean;
+
+        /**  Whether the attachment is deleted */
```
Comment (mananjadhav) [inline src/components/Attachments/AttachmentView/index.tsx:81]:
```suggestion
        /** Whether the attachment is deleted */
```

File: src/components/Attachments/AttachmentView/DefaultAttachmentView/index.tsx [hunk 4]
```diff
@@ -24,9 +24,12 @@ type DefaultAttachmentViewProps = {
     containerStyles?: StyleProp<ViewStyle>;
 
     icon?: IconAsset;
+
+    /**  Whether the attachment is deleted */
```
Comment (mananjadhav) [inline src/components/Attachments/AttachmentView/DefaultAttachmentView/index.tsx:28]:
```suggestion
    /** Whether the attachment is deleted */
```

File: src/components/VideoPlayerPreview/VideoPlayerThumbnail.tsx [hunk 5]
```diff
@@ -22,9 +23,12 @@ type VideoPlayerThumbnailProps = {
 
     /** Accessibility label for the thumbnail. */
     accessibilityLabel: string;
+
+    /**  Whether the video is deleted */
```
Comment (mananjadhav) [inline src/components/VideoPlayerPreview/VideoPlayerThumbnail.tsx:27]:
```suggestion
    /** Whether the video is deleted */
```

File: src/components/VideoPlayerPreview/VideoPlayerThumbnail.tsx [hunk 6]
```diff
@@ -39,35 +43,39 @@ function VideoPlayerThumbnail({thumbnailUrl, onPress, accessibilityLabel}: Video
...
+                                    width={variables.iconSizeXLarge}
+                                    height={variables.iconSizeXLarge}
+                                    additionalStyles={[styles.ml1]}
+                                />
+                            </View>
+                        </PressableWithoutFeedback>
+                    )}
+                </ShowContextMenuContext.Consumer>
+            ) : (
+                <AttachmentDeletedIndicator containerStyles={{borderRadius: variables.componentBorderRadiusNormal}} />
```
Comment (mananjadhav) [inline src/components/VideoPlayerPreview/VideoPlayerThumbnail.tsx:77]: Let's move this to styles?

File: src/components/VideoPlayerPreview/index.tsx [hunk 7]
```diff
@@ -38,9 +38,12 @@ type VideoPlayerPreviewProps = {
 
     /** Callback executed when modal is pressed. */
     onShowModalPress: (event?: GestureResponderEvent | KeyboardEvent) => void | Promise<void>;
+
+    /**  Whether the video is deleted */
```
Comment (mananjadhav) [inline src/components/VideoPlayerPreview/index.tsx:42]:
```suggestion
    /** Whether the video is deleted */
```

File: src/components/VideoPlayerPreview/index.tsx [hunk 8]
```diff
@@ -71,11 +74,12 @@ function VideoPlayerPreview({videoUrl, thumbnailUrl, reportID, fileName, videoDi
 
     return (
         <View style={[styles.webViewStyles.tagStyles.video, thumbnailDimensionsStyles]}>
-            {shouldUseNarrowLayout || isThumbnail ? (
+            {shouldUseNarrowLayout || isThumbnail || isDeleted ? (
```
Comment (mananjadhav) [inline src/components/VideoPlayerPreview/index.tsx:77]: Do we need the `isDeleted` check in the `||` chain?

File: src/pages/home/report/comment/AttachmentCommentFragment.tsx [hunk 9]
```diff
@@ -14,8 +13,7 @@ type AttachmentCommentFragmentProps = {
 
 function AttachmentCommentFragment({addExtraMargin, html, source, styleAsDeleted}: AttachmentCommentFragmentProps) {
     const styles = useThemeStyles();
-    const isUploading = html.includes(CONST.ATTACHMENT_OPTIMISTIC_SOURCE_ATTRIBUTE);
```
Comment (mananjadhav) [inline src/pages/home/report/comment/AttachmentCommentFragment.tsx:17]: I missed this why is `isUploading` removed?

File: src/styles/index.ts [hunk 10]
```diff
@@ -1111,6 +1111,17 @@ const styles = (theme: ThemeColors) =>
             height: 25,
         },
 
+        deletedIndicator: {
```
Comment (mananjadhav) [inline src/styles/index.ts:1114]:
```suggestion
        deletedAttachmentIndicator: {
```

File: src/components/AttachmentDeletedIndicator.tsx [hunk 11]
```diff
@@ -0,0 +1,44 @@
...
+import type {StyleProp, ViewStyle} from 'react-native';
+import useNetwork from '@hooks/useNetwork';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+
+type AttachmentOfflineIndicatorProps = {
+    /** Any additional styles to apply */
```
Comment (mananjadhav) [inline src/components/AttachmentDeletedIndicator.tsx:12]:
```suggestion
    /** Additional styles for container */
```

File: src/components/AttachmentDeletedIndicator.tsx [hunk 12]
```diff
@@ -0,0 +1,44 @@
...
+import {View} from 'react-native';
+import type {StyleProp, ViewStyle} from 'react-native';
+import useNetwork from '@hooks/useNetwork';
+import useTheme from '@hooks/useTheme';
+import useThemeStyles from '@hooks/useThemeStyles';
+import variables from '@styles/variables';
+import Icon from './Icon';
+import * as Expensicons from './Icon/Expensicons';
+
+type AttachmentOfflineIndicatorProps = {
```
Comment (mananjadhav) [inline src/components/AttachmentDeletedIndicator.tsx:11]:
```suggestion
type AttachmentDeletedIndicatorProps = {
```

File: src/components/AttachmentDeletedIndicator.tsx [hunk 13]
```diff
@@ -0,0 +1,44 @@
...
+    /** Any additional styles to apply */
+    containerStyles?: StyleProp<ViewStyle>;
+};
+
+function AttachmentDeletedIndicator({containerStyles}: AttachmentOfflineIndicatorProps) {
```
Comment (mananjadhav) [inline src/components/AttachmentDeletedIndicator.tsx:16]:
```suggestion
function AttachmentDeletedOfflineIndicator({containerStyles}: AttachmentOfflineIndicatorProps) {
```

File: src/components/AttachmentDeletedIndicator.tsx [hunk 14]
```diff
@@ -0,0 +1,44 @@
...
+    const {isOffline} = useNetwork();
+
+    if (!isOffline) {
+        return null;
+    }
+
+    return (
+        <>
+            <View
+                style={[styles.pAbsolute, styles.alignItemsCenter, styles.justifyContentCenter, styles.highlightBG, styles.deletedIndicatorOverlay, styles.deletedIndicator, containerStyles]}
```
Comment (mananjadhav) [inline src/components/AttachmentDeletedIndicator.tsx:28]: Still going through these styles.

**Review by mananjadhav — COMMENTED** [review]
Left a few comments.

**Comment by neil-marcellini** [comment]
> @Krishna2323 I think we'll need to fix the eslint issues even if they're unrelated to the PR. It is because we're touching those files and the new lint rules are failing.
> 
> @neil-marcellini can you please confirm?

Hi, thanks for asking. The latest I've heard is that if the ES Lint checks are related to files touched in the current PR, then we require them to be fixed before the issue can be considered complete and paid. If the fixes are small then we like to see the changes in the current PR. Otherwise, like in this case where we need to migrate to useOnyx, we should handle it in a separate PR. You could also search to see if there is already an issue to migrate to useOnyx for these files.

**Comment by mananjadhav** [comment]
I couldn't check this over the weekend, but I'll try to check today. I think it's best to post on open-source channel to get more eyes.

**Comment by mananjadhav** [comment]
Thanks for the update @Krishna2323. I will check this out today.

**Comment by mananjadhav** [comment]
## Reviewer Checklist

- [x] I have verified the author checklist is complete (all boxes are checked off).
- [x] I verified the correct issue is linked in the `### Fixed Issues` section above
- [x] I verified testing steps are clear and they cover the changes made in this PR
    - [x] I verified the steps for local testing are in the `Tests` section
    - [x] I verified the steps for Staging and/or Production testing are in the `QA steps` section
    - [x] I verified the steps cover any possible failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
- [x] I checked that screenshots or videos are included for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [ ] I verified tests pass on **all platforms** & I tested again on:
    - [ ] Android: Native
    - [ ] Android: mWeb Chrome
    - [ ] iOS: Native
    - [ ] iOS: mWeb Safari
    - [ ] MacOS: Chrome / Safari
    - [ ] MacOS: Desktop
- [ ] If there are any errors in the console that are unrelated to this PR, I either fixed them (preferred) or linked to where I reported them in Slack
- [x] I verified proper code patterns were followed (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`).
    - [x] I verified that the left part of a conditional rendering a React component is a boolean and NOT a string, e.g. `myBool && <MyComponent />`.
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is either coming verbatim from figma or has been approved by marketing (in order to get marketing approval, ask the Bug Zero team member to add the Waiting for copy label to the issue)
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I verified that this PR follows the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I verified other components that can be impacted by these changes have been tested, and I retested again (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` have been tested & I retested again)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/utils/index.ts) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(theme.componentBG`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If the PR modifies a component or page that can be accessed by a direct deeplink, I verified that the code functions as expected when the deeplink is used - from a logged in and logged out account.
- [x] If the PR modifies the UI (e.g. new buttons, new UI components, changing the padding/spacing/sizing, moving components, etc) or modifies the form input styles:
    - [x] I verified that all the inputs inside a form are aligned with each other.
    - [x] I added `Design` label and/or tagged `@Expensify/design` so the design team can review the changes.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [ ] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [ ] I have checked off every checkbox in the PR reviewer checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Android: Native</summary>



</details>

<details>
<summary>Android: mWeb Chrome</summary>



</details>

<details>
<summary>iOS: Native</summary>



</details>

<details>
<summary>iOS: mWeb Safari</summary>



</details>

<details>
<summary>MacOS: Chrome / Safari</summary>



</details>

<details>
<summary>MacOS: Desktop</summary>



</details>

**Comment by mananjadhav** [comment]
Did the latest review, left with the QA on mobile devices. Will finish the checklist today.

**Review by mananjadhav — APPROVED** [review]
I am having issues with iOS so skipping that one. Screenshots for the rest are uploaded and I verified iOS on the author checklist.

**Review by neil-marcellini — APPROVED** [review]
The code looks fine and looks like it tests well. Thank you for the hard work.

**Pull request merged**